Source/missiles.cpp

```diff
@@ -915,6 +874,105 @@ Direction16 GetDirection16(Point p1, Point p2)
 	return ret;
 }
 
+void MissileHitMonsterConsequences(int mid, int pnum, int dam, missile_id mName)
+{
+	auto &monster = Monsters[mid];
+	const auto &player = Players[pnum];
+
+	bool hasKnockback = false;
+	if (pnum >= 0) {
+		const auto &player = Players[pnum];
+		if (HasAnyOf(player._pIFlags, ItemSpecialEffect::Knockback))
+			hasKnockback = true;
+	}
+
+	if (monster._mhitpoints >> 6 <= 0) {
+		M_StartKill(mid, pnum);
+	} else if (monster.IsResistant(mName)) {
+		PlayEffect(monster, 1);
+	} else {
+		if (monster._mmode != MonsterMode::Petrified
+		    && MissilesData[mName].mType == 0
+		    && hasKnockback)
+			M_GetKnockback(mid);
+		if (monster.MType->mtype != MT_GOLEM)
+			M_StartHit(mid, pnum, dam);
+	}
+}
+
+int PlayerMissileCalculateCTHAgainstMonster(int pnum, missile_id mName, int dist, Monster &monster)
+{
+	int hper = 0;
+	if (pnum != -1) {

```

Comment:
When you have a special case like this it's preferable to do an early return.
```cpp
if (pnum == -1) {
    return player.GetMagicToHit() - (monster.mLevel * 2) - dist;
}
```

But in this case I would actually suggest that you move the case out of the function so that the function can take `Player &player` as it's input rather then int `pnum`.

Reply:
yes, I agree to early return, but I am not sure is it even used at all :joy: 

as for parameter list - it will be changed in 2 functions merge to 1 commit in this pr.

Reply:
this is probably resolved now